### PR TITLE
feat: Add yaml generation script

### PR DESCRIPTION
**Pull request by swetshaw**
This PR adds scripts for generating the chains yaml file after deploying the contracts

### CODE REVIEW

File: src/utils/gen-chain-yaml.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    // Read yaml file from inside chains folder
+    const yamlFilePath = join(chainsDirectory, `${chainName}.yaml`)
+    let yaml = await readYaml<ChainData>(yamlFilePath)
+
+    // Write the values of contract addresses from config to the yaml file
+    const qbContracts = {
+        "applications": {
+            "address": config.applicationRegistryAddress.proxy,
+            "startBlock": 1,
```
Comment (adiwajshing) [inline src/utils/gen-chain-yaml.ts:22]: We def need the right start block otherwise the subgraph will take forever to sync

Comment (swetshaw) [reply src/utils/gen-chain-yaml.ts:22]: Yes, will update the script in the contracts repo to get the startBlock numbers.

File: README.md [hunk 2]
```diff
@@ -5,11 +5,16 @@ Single source of truth for all the chains supported by the QuestBook app &amp; t
...
-3. Create a new yaml file in the `./chains` with the name of the chain. For eg. if your chain is the testnet for XYZ, then the file would be called `xyz-testnet.yaml`
-4. Enter all the required configuration, as mentioned in the JSON schema stored at `./src/chain-data-schema.yaml`.
-	- This would include details like chain name, ID, addresses of contracts deployed, block number of the contract deployment, etc.
+2. Create a new yaml file in the `./chains` with the name of the chain. For eg. if your chain is the testnet for XYZ, then the file would be called `xyz-testnet.yaml`
+3. Enter the following information:
+    - chainName
+    - chainId
+    - rpcUrls
+4. Run `NETWORK={$NETWORK} yarn:add chain` to add your chain to our official list of networks. Note: The value of `$NETWORK` should be same as the name of your chains yaml file. For example, `NETWORK=xyz-testnet yarn:add chain`
+9. This will add the contract addresses to your chain's yaml file.
```
Comment (adiwajshing) [inline README.md:14]: numbers need to be updated

File: package.json [hunk 3]
```diff
@@ -8,7 +8,8 @@
     "start:tsc": "ts-node src",
     "lint": "eslint ./src --ext .js,.ts,.jsx,.tsx",
     "lint:fix": "eslint ./src --fix --ext .js,.ts,.jsx,.tsx",
-    "gen:types": "ts-node src/utils/gen-types"
+    "gen:types": "ts-node src/utils/gen-types",
+    "add:chain": "ts-node src --dry-run --repo grants-contracts-upgradeable && cd contracts/evm && yarn && yarn typechain && NETWORK=$NETWORK npm run deploy && cd ../.. && ts-node src/utils/gen-chain-yaml --chain-name $NETWORK"
```
Comment (adiwajshing) [inline package.json:12]: the dry run will commit to the grants-contracts-upgradeable repo in the tmp folder, not our submodule

**Review by adiwajshing — APPROVED** [review]

**Pull request merged**